lib/db/memory.js

```diff
@@ -269,7 +270,14 @@ MemoryStore.prototype = {
       if (client.canGrant === false) {
         if (! activeClients[clientIdHex]) {
           activeClients[clientIdHex] = client;
+          activeClients[clientIdHex].scope = [];
         }
+        clientTokenObj.clientScope.forEach(function (scope) {
```

Comment:
Oh sorry I got confused with the way scopes are stored. Let's add a test for this new behaviour. 
We need the test to issue ~3-4 tokens with different scopes and return the unified list.